pkg/ldapsession/search.go

```diff
@@ -43,7 +44,24 @@ func (w *LDAPSession) ManualWriteSearchResultsToChan(results *ldap.SearchResult)
 	for _, control := range results.Controls {
 		w.Channels.Controls <- control
 	}
+}
+
+func (w *LDAPSession) ManualWriteMultipleSearchResultsToChan(multipleResults []*ldap.SearchResult) {
```

Comment:
Ugh, sorry you had to do this. I get why - the existing ManualWrite closes the channels when it returns. This totally works but let me think if there's a better way I can make this easier for future modules that need to make multiple requests

Reply:
I went for the straight forward approach, but I'm open to your suggestions for a better approach

Reply:
Yep, I'm working on adding an `ExecuteBulkSearchRequest` function that takes in `[]*ldap.SearchRequest` and executes them sequentially, writing all results to the same channels. That should be easier and allow just the module logic to construct whatever search requests it wants executed and then execute them all at once